src/serialize.h

```diff
@@ -1142,6 +1146,24 @@ class ParamsStream
             return m_substream.template GetParams<P>();
         }
     }
+
+    //! Get reference to underlying stream.
+    auto& GetStream()
+    {
+        if constexpr (ContainsStream<SubStream>) {
+            return m_substream.GetStream();
```

Comment:
Just a comment: This does indeed not trigger clang-tidy's recursion check (I'm guessing because of the constexpr condition).